tokio/src/sync/mutex.rs

```diff
@@ -86,11 +6,89 @@ use std::error::Error;
 use std::fmt;
 use std::ops::{Deref, DerefMut};
 
-/// An asynchronous mutual exclusion primitive useful for protecting shared data
+/// An asynchronous `Mutex`-like type.
 ///
-/// Each mutex has a type parameter (`T`) which represents the data that it is protecting. The data
-/// can only be accessed through the RAII guards returned from `lock`, which
-/// guarantees that the data is only ever accessed when the mutex is locked.
+/// This type acts similarly to an asynchronous [`std::sync::Mutex`], with one
+/// major difference: [`lock`] does not block.\
+/// Another difference is that [`MutexGuard`] is [`Send`]
```

Comment:
I think perhaps this should just say that `lock` can be held across await points? Fix also the backlash.